### PR TITLE
:sparkles: [Appli - CGS] Prepare route to redirect to cgs post login

### PR DESCRIPTION
**Pull request by tcheymol**
[(8) ETQU, je dois accepter les nouvelles CGU, CGS, politiques de confidentialité,... [ATTENTE FICHIER CGS]](https://trello.com/c/x6jm180H/5008-8-etqu-je-dois-accepter-les-nouvelles-cgu-cgs-politiques-de-confidentialit%C3%A9-attente-fichier-cgs)

### CODE REVIEW

File: src/EventSubscriber/Api/Oauth2ResponseSubscriber.php [hunk 1]
```diff
@@ -100,4 +93,14 @@ public function extractUserIdFromJwt(string $jwt): string
...
     }
+
+    private function findUser(TokenRequestResolveEvent $event): ?User
+    {
+        $username = $this->extractUserIdFromResponse($event->getResponse());
+        if (!$username) {
+            return null;
+        }
+
+        return $this->repository->findByUsername($username);
```
Comment (AntoineAresu) [inline src/EventSubscriber/Api/Oauth2ResponseSubscriber.php:104]: `return username ? $this->repository->findByUsername($username) : null;`

File: src/Entity/User.php [hunk 2]
```diff
@@ -1307,6 +1307,11 @@ public function getCgsAcceptedAt(): ?\DateTimeImmutable
         return $this->cgsAcceptedAt;
     }
 
+    public function hasAcceptedCgs(): bool
+    {
+        return true;
+    }
+
```
Comment (AntoineAresu) [inline src/Entity/User.php:1314]: I think there is a problem

Comment (AntoineAresu) [reply src/Entity/User.php:1314]: If it's on purpose for feature preparation, maybe a comment would be helpfull

File: src/EventSubscriber/Api/Oauth2ResponseSubscriber.php (hunk 1 above)
Comment (tcheymol) [reply src/EventSubscriber/Api/Oauth2ResponseSubscriber.php:104]:
```php
    public function findByUsername($username): ?User
    {
        $query = $this->createQueryBuilder('user')
            ->where('user.username = :username')
            ->setParameter('username', $username)
            ->getQuery();

        try {
            $user = $query->getOneOrNullResult();
        } catch (NonUniqueResultException) {
            return null;
        }

        return $user;
    }
```

File: src/ServiceV2/GdprService.php [hunk 3]
```diff
@@ -29,6 +30,11 @@ public function isPasswordRenewalDue(): bool
         return $this->getDaysBeforeExpiration() <= self::RENEWAL_DAYS_COUNT;
     }
 
+    public function shouldRenewPassword(?User $user = null): bool
```
Comment (NilsRec) [inline src/ServiceV2/GdprService.php:33]: `mustRenewPassword` pour être + cohérent avec `mustAcceptTermsOfUse` ?

**Review by NilsRec — APPROVED** [review]

File: src/Domain/PasswordStrength/WeakPasswordUpgrader.php [hunk 4]
```diff
@@ -30,4 +30,12 @@ public function checkUpdateWeakPassword(User $user, mixed $newPassword): void
...
         }
     }
+
+    public function checkAcceptTermsOfUse(User $user, ?bool $shouldAccept = null): void
+    {
+        if (!$shouldAccept) {
+            $user->acceptTermsOfUse();
+            $this->em->flush($user);
+        }
+    }
```
Comment (AntoineAresu) [inline src/Domain/PasswordStrength/WeakPasswordUpgrader.php:40]: I think it's in a wrong namespace

File: src/Entity/User.php [hunk 5]
```diff
@@ -1314,6 +1313,12 @@ public function setCgsAcceptedAt(?\DateTimeImmutable $cgsAcceptedAt): static
         return $this;
     }
 
+    public function acceptTermsOfUse(): static {
+        $this->cgsAcceptedAt = new \DateTimeImmutable();
+
+        return $this;
+    }
+
```
Comment (AntoineAresu) [inline src/Entity/User.php:1321]:
```
    public function acceptTermsOfUse(): static 
    {
        $this->cgsAcceptedAt = new \DateTimeImmutable();

        return $this;
    }
```

File: src/EventSubscriber/Api/Oauth2ResponseSubscriber.php (hunk 1 above)
Comment (AntoineAresu) [reply src/EventSubscriber/Api/Oauth2ResponseSubscriber.php:106]: Et du coup ?

**Review by NilsRec — APPROVED** [review]

**Pull request merged**